lib/location/method_gnss.c

```diff
@@ -41,11 +41,11 @@ BUILD_ASSERT(
 	"CONFIG_LOCATION_SERVICE_EXTERNAL must be enabled");
 #endif
 
-/* Maximum waiting time before GNSS is started regardless of RRC or PSM state [min]. This prevents
+/* Maximum waiting time before proceeding regardless of RRC or PSM state [min]. This prevents
  * Location library from getting stuck indefinitely if the application keeps LTE connection
  * constantly active.
  */
-#define SLEEP_WAIT_BACKSTOP 5
+#define SLEEP_WAIT_BACKSTOP 2
```

Comment:
Just to make sure you understand the change. This timeout value is for waiting for RRC idle always when GNSS fix is requested but it also applies to PSM waiting when A-GNSS is disabled.

2 minutes should be fine for RRC idle mode waiting time but is this enough for PSM waiting time?
I'm trying to find values with which we would still get fallbacks to work with default 5min location request timeout. And I realize it may not be possible with A-GNSS disabled when we wait both for RRC idle and PSM.

Reply:
The idea behind the selected value has been that it should be long enough not to cause problems in cases where the RRC connection release or entering PSM takes longer, but to prevent the GNSS method from waiting forever.

Both cases currently use the same timeout, but it does not necessarily have to be like that. Starting from RRC connection release, there's not much point in starting GNSS before the RRC connection release, because GNSS will not get runtime and can not get a fix. So, I think we could even wait for RRC connection release until the GNSS method timeout.

Waiting for PSM is different. We only do that when A-GNSS is disabled. The idea is, that getting a fix in RRC idle is not possible, because GNSS can not decode data from the satellite broadcast. That's not the whole story, though. First, GNSS might not need to decode any data to get a fix if it's hot enough. And secondly, we do not take eDRX into account. If both PSM and eDRX are configured, it would be possible for GNSS to get enough runtime already during PSM active time (depending on the eDRX cycle).

The problem with waiting for PSM is that we don't know for sure if it's necessary or not. We could add another check for eDRX, but we don't know if GNSS needs to decode ephemerides or TOW to get a fix. It's not very likely that GNSS has all the data it needs without A-GNSS, I guess. Delaying GNSS start when getting a fix is not possible is beneficial, because GNSS would get short time windows during legacy DRX and unnecessarily consume power.

@hiltunent Any thoughts?

Reply:
Decided to keep the implementation as is after some discussions.